engine/lib/phx/script/ffi_ext/EventBus.lua

```diff
@@ -0,0 +1,17 @@
+local libphx = require('libphx').lib
+
+function onDef_EventBus_t(t, mt)
+    -- todo add LS definitions
+    -- todo should also return a handler
+    mt.__index.subscribe = function(self, eventName, ctxTable, callback)
+        local entityId = ctxTable and ctxTable.getGuid and ctxTable.getGuid()
+        local tunnelId = libphx.EventBus_Subscribe(self, eventName, entityId)
```

Comment:
LuaJIT does have first class support for callbacks: 

Which means that in theory, we could pass `function() callback(ctxTable) end` directly to `EventBus_Subscribe`, and make the 4th parameter `callback: fn() -> ()`. The problem is that luajit-ffi-gen doesn't support that yet.

Reply:
cc @HaronK

Reply:
I think if we could somehow utilise this callback system, then that would allow us to avoid having to pass around the Lua state in the `dispatch` method.

Reply:
I can think about adding support for such a case.

Reply:
Btw, from the link above: **Callbacks are slow!**